pulser/tests/test_simulation.py

```diff
@@ -249,61 +261,139 @@ def test_run():
+        sim.evaluation_times = 'Best'
 
     with pytest.raises(ValueError,
                        match="Provided evaluation-time list contains "
                              "negative values."):
-        sim = Simulation(seq, sampling_rate=1.,
-                         evaluation_times=[-1, 0, sim._times[-2]])
+        sim = Simulation(seq, sampling_rate=1.)
+        sim.evaluation_times = [-1, 0, sim._times[-2]]
 
     with pytest.raises(ValueError,
                        match="Provided evaluation-time list extends "
                              "further than sequence duration."):
-        sim = Simulation(seq, sampling_rate=1.,
-                         evaluation_times=[0, sim._times[-1]+10])
+        sim = Simulation(seq, sampling_rate=1.)
+        sim.evaluation_times = [0, sim._times[-1]+10]
 
-    sim = Simulation(seq, sampling_rate=1., evaluation_times='Full')
-    np.testing.assert_almost_equal(sim.eval_times, sim._times)
+    sim = Simulation(seq, sampling_rate=1.)
+    sim.evaluation_times = 'Full'
+    assert sim.evaluation_times == 'Full'
+    np.testing.assert_almost_equal(sim._eval_times_array, sim._times)
 
-    sim = Simulation(seq, sampling_rate=1., evaluation_times='Minimal')
-    np.testing.assert_almost_equal(sim.eval_times,
+    sim = Simulation(seq, sampling_rate=1.)
+    sim.evaluation_times = 'Minimal'
+    np.testing.assert_almost_equal(sim._eval_times_array,
                                    np.array([sim._times[0], sim._times[-1]])
                                    )
 
-    sim = Simulation(seq, sampling_rate=1.,
-                     evaluation_times=[0, sim._times[-3], sim._times[-1]])
-    np.testing.assert_almost_equal(sim.eval_times,
+    sim = Simulation(seq, sampling_rate=1.)
+    sim.evaluation_times = [0, sim._times[-3], sim._times[-1]]
+    np.testing.assert_almost_equal(sim._eval_times_array,
                                    np.array([0, sim._times[-3],
                                              sim._times[-1]])
                                    )
 
-    sim = Simulation(seq, sampling_rate=1.,
-                     evaluation_times=[sim._times[-10], sim._times[-3]])
-    np.testing.assert_almost_equal(sim.eval_times,
+    sim = Simulation(seq, sampling_rate=1.)
+    sim.evaluation_times = [sim._times[-10], sim._times[-3]]
+    np.testing.assert_almost_equal(sim._eval_times_array,
                                    np.array([0, sim._times[-10],
                                              sim._times[-3], sim._times[-1]])
                                    )
+
+    sim = Simulation(seq, sampling_rate=1.)
+    sim.evaluation_times = 0.4
+
+
+def test_config():
+    sim = Simulation(seq, sampling_rate=0.01)
+    sim.reset_config()
+    assert sim.config == SimConfig()
+    sim.show_config()
+    with pytest.raises(ValueError, match="not a valid"):
+        sim.set_config('bad_config')
+    new_cfg = SimConfig(noise='SPAM')
+    sim.set_config(new_cfg)
+    assert sim.config == new_cfg
```

Comment:
Maybe you can store the Hamiltonian objects and assert that they change (or not change) like you expect them to when you change the config.